server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -542,6 +564,73 @@ public Consumer getHost(String guestId, Owner... owners) {
         return host;
     }
 
+    /**
+     * Creates a mapping of input guest IDs to GuestID objects currently tracked and stored in the
+     * backing database. If a given guest ID is not present in the database, it will be mapped to
+     * a null value.
+     *
+     * @param guestIds
+     *  A collection of guest IDs to map to existing GuestID objects
+     *
+     * @param owner
+     *  The owner to which the mapping lookup should be scoped
+     *
+     * @return
+     *  a mapping of guest IDs to GuestID objects
+     */
+    @Transactional
+    public Map<String, GuestId> getGuestIdMap(Iterable<String> guestIds, Owner owner) {
+        if (guestIds == null || owner == null) {
+            return Collections.<String, GuestId>emptyMap();
+        }
+
+        String hql = "SELECT gid FROM GuestId gid " +
+            "WHERE gid.consumer.owner = :owner AND gid.guestIdLower IN (:guest_ids)" +
+            "ORDER BY gid.updated DESC";
+
+        TypedQuery<GuestId> query = this.getEntityManager().createQuery(hql, GuestId.class);
+        Map<String, GuestId> output = new HashMap<String, GuestId>();
+
+        for (List<String> block : this.partition(guestIds)) {
+            List<String> sanitized = new ArrayList<String>(block);
```

Comment:
Whoops... made a mistake here. That needs to be block.size(), not block. As-is, we'll double our block size and potentially exceed our param limit here.